documentation/src/main/asciidoc/topics/search.adoc

```diff
@@ -0,0 +1,1087 @@
+executed, calling `run()` on the same instance will re-execute the query.
+
+=== Pagination
+
+You can limit the number of returned results by using  the `Query.maxResults(int maxResults)`. This can be used in
+conjunction with `Query.startOffset(long startOffset)` to achieve pagination of the result set.
+
+[source,java]
+----
+// sorted by year and match all books that have "clustering" in their title
+// and return the third page of 10 results
+Query<Book> query = queryFactory.create("FROM com.acme.Book WHERE title like '%clustering%' ORDER BY year").startOffset(20).maxResults(10)
+----
+
+=== Number of Hits
+
+The `QueryResult` object has the `.hitCount()` method to return the total number of results of the query, regardless
+of any pagination parameter. The hit count is only available for indexed queries for performance reasons.
+
+
+=== Iteration
+
+The `Query` object has the `.iterator()` method to obtain the results lazily. It returns an instance of `CloseableIterator` that must be closed after usage.
+
+[NOTE]
+====
+The iteration support for Remote Queries is currently limited, as it will first fetch all entries to the client
+before iterating.
+====
+
+=== Using Named Query Parameters
+
+Instead of building a new Query object for every execution it is possible to include named parameters in the query which
+can be substituted with actual values before execution. This allows a query to be defined once and be efficiently
+executed many times. Parameters can only be used on the right-hand side of an operator and are defined when the query is
+created by supplying an object produced by the _org.infinispan.query.dsl.Expression.param(String paramName)_ method to
+the operator instead of the usual constant value. Once the parameters have been defined they can be set by invoking either
+_Query.setParameter(parameterName, value)_ or _Query.setParameters(parameterMap)_ as shown in the examples below.
+⁠
+[source,java,tile="Using Named Parameters"]
+----
+QueryFactory queryFactory = Search.getQueryFactory(cache);
+// Defining a query to search for various authors and publication years
+Query<Book> query = queryFactory.create("SELECT title FROM com.acme.Book WHERE author = :authorName AND publicationYear = :publicationYear").build();
+
+// Set actual parameter values
+query.setParameter("authorName", "Doe");
+query.setParameter("publicationYear", 2010);
+
+// Execute the query
+List<Book> found = query.list();
+----
+
+Alternatively, you can supply a map of actual parameter values to set multiple parameters at once:
+⁠
+[source,java,title="Setting multiple named parameters at once"]
+----
+Map<String, Object> parameterMap = new HashMap<>();
+parameterMap.put("authorName", "Doe");
+parameterMap.put("publicationYear", 2010);
+
+query.setParameters(parameterMap);
+----
+
+NOTE: A significant portion of the query parsing, validation and execution planning effort is performed during the first
```

Comment:
Notes should all use the same format.

```
[NOTE]
====
Read this note.
====
```

Reply:
right, there are quite a few NOTEs that need be upgraded, I will change all